src/components/tl/ucp/tl_ucp_copy.c

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_ucp.h"
+#include "tl_ucp_coll.h"
+#include "tl_ucp_tag.h"
+#include "tl_ucp_ep.h"
+#include "tl_ucp_sendrecv.h"
+#include "tl_ucp_copy.h"
+
+ucc_status_t ucc_tl_ucp_mc_copy_post(void *dst, ucc_memory_type_t dst_mtype,
+                                     void *src, ucc_memory_type_t src_mtype,
+                                     size_t size,
+                                     ucc_tl_ucp_task_t *coll_task, //NOLINT
+                                     ucc_tl_ucp_copy_task_t **copy_task) //NOLINT
+{
+    return ucc_mc_memcpy(dst, src, size, dst_mtype, src_mtype);
+}
+
+ucc_status_t ucc_tl_ucp_mc_copy_test(ucc_tl_ucp_context_t *ctx, //NOLINT
+                                     ucc_tl_ucp_copy_task_t *copy_task) //NOLINT
+{
+    /* mc copy is blocking, test always returns UCC_OK */
+    return UCC_OK;
+}
+
+ucc_status_t ucc_tl_ucp_mc_copy_finalize(ucc_tl_ucp_copy_task_t *copy_task) //NOLINT
+{
+    return UCC_OK;
+}
+
+ucc_status_t ucc_tl_ucp_ec_copy_post(void *dst, ucc_memory_type_t dst_mtype, //NOLINT
+                                     void *src, ucc_memory_type_t src_mtype, //NOLINT
+                                     size_t size,
+                                     ucc_tl_ucp_task_t *coll_task,
+                                     ucc_tl_ucp_copy_task_t **copy_task)
+{
+    ucc_ee_executor_task_args_t eargs = {0};
+    ucc_ee_executor_task_t     **eee_task = (ucc_ee_executor_task_t **)copy_task;
+    ucc_status_t status;
+    ucc_ee_executor_t *exec;
+
+    status = ucc_coll_task_get_executor(&coll_task->super, &exec);
+    if (ucc_unlikely(status != UCC_OK)) {
+        return status;
+    }
+
+    eargs.task_type = UCC_EE_EXECUTOR_TASK_COPY;
+    eargs.copy.dst  = dst;
+    eargs.copy.src  = src;
+    eargs.copy.len  = size;
+
+    return ucc_ee_executor_task_post(exec, &eargs, eee_task);
+}
+
+ucc_status_t ucc_tl_ucp_ec_copy_test(ucc_tl_ucp_context_t *ctx, //NOLINT
+                                     ucc_tl_ucp_copy_task_t *copy_task)
+{
+    ucc_ee_executor_task_t *eee_task = (ucc_ee_executor_task_t *)copy_task;
+
+    return ucc_ee_executor_task_test(eee_task);
+}
+
+ucc_status_t ucc_tl_ucp_ec_copy_finalize(ucc_tl_ucp_copy_task_t *copy_task)
+{
+    ucc_ee_executor_task_t *eee_task = (ucc_ee_executor_task_t *)copy_task;
```

Comment:
Here is a clarification on what I mentionned during our discussion. I would personally prefer
```suggestion
    ucc_ee_executor_task_t *eee_task = &copy_task->ee_task;
```
which avoids manually casting without any compiler check. To make this last statement clearer, here's a snippet illustrating what I mean: 
```
typedef struct a {
    int i;
} a_t;

typedef struct b {
    int j;
} b_t;

typedef union union_test {
    a_t a;
    b_t b;
} union_test_t;

int main() {
    union_test_t x;
    
    a_t y = x.a;
    // int z = x.a; invalid cast that is detected by the compile -- this won't compile
    int z = *(int*)&x.a; // invalid cast that is NOT detected by the compiler -- this will compile

    return 0;
}
```